mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -221,159 +224,212 @@ abstract class PcCollector[T](
+          f(sought)
+        }
 
-            /**
-             * We don't automatically traverser types like:
-             * val opt: Option[<<String>>] =
-             */
-            case tpe: TypeTree if tpe.original != null =>
-              tpe.original.children.foldLeft(acc)(traverse(_, _))
-            /**
-             * Some type trees don't have symbols attached such as:
-             * type A = List[_ <: <<Iterable>>[Int]]
-             */
-            case id: Ident
-                if id.symbol == NoSymbol && soughtNames.exists(_ == id.name) =>
-              fallbackSymbol(id.name, id.pos) match {
-                case Some(sym) if sought(sym) =>
-                  acc + collect(
-                    id,
-                    id.pos
-                  )
-                case _ => acc
-              }
+        traverseSought(soughtTreeFilter, soughtFilter).toList
 
-            /**
-             * For traversing annotations:
-             * @<<JsonNotification>>("")
-             * def params() = ???
-             */
-            case df: MemberDef =>
-              (tree.children ++ annotationChildren(df))
-                .foldLeft(acc)(traverse(_, _))
+      case None => Nil
+    }
+  }
+
+  def resultAllOccurences(): Set[T] = {
+    def noTreeFilter = (tree: Tree) => true
+    def noSoughtFilter = (f: Set[Symbol] => Boolean) => true
 
-            /**
-             * For traversing import selectors:
-             * import scala.util.<<Try>>
-             */
-            case imp: Import
-                if owners(imp.expr.symbol) && imp.selectors
-                  .exists(sel => soughtNames(sel.name)) =>
-              imp.selectors.foldLeft(traverse(acc, imp.expr)) {
-                case (acc, sel) if soughtNames(sel.name) =>
-                  val positions =
-                    if (!sel.rename.isEmpty)
-                      Set(
-                        sel.renamePosition(pos.source),
-                        sel.namePosition(pos.source)
-                      )
-                    else Set(sel.namePosition(pos.source))
-                  acc ++ positions.map(pos => collect(imp, pos))
-                case (acc, _) =>
-                  acc
+    traverseSought(noTreeFilter, noSoughtFilter)
+  }
+
+  def traverseSought(
+      filter: Tree => Boolean,
+      soughtFilter: (Set[Symbol] => Boolean) => Boolean
+  ): Set[T] = {
+    // Now find all matching symbols in the document, comments identify <<>> as the symbol we are looking for
+    def traverse(
+        acc: Set[T],
+        tree: Tree
+    ): Set[T] = {
+      tree match {
+        /**
+         * All indentifiers such as:
+         * val a = <<b>>
+         */
+        case ident: Ident if ident.pos.isRange && filter(ident) =>
+          val id =
+            if (ident.symbol == NoSymbol) {
+              fallbackSymbol(ident.name, pos) match {
+                case None => ident
+                case Some(sym) => Ident(sym)
```

Comment:
This is copied from `soughtSymbols`. It typed trees we get `ident.symbol == NoSymbol`, so we use fallback method. It worked similarly in `SemanticTokenProvider` traverser